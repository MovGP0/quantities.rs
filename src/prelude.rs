// ---------------------------------------------------------------------------
// Copyright:   (c) 2021 ff. Michael Amrhein (michael@adrhinum.de)
// License:     This program is part of a larger application. For license
//              details please read the file LICENSE.TXT provided together
//              with the application.
// ---------------------------------------------------------------------------
// $Source$
// $Revision$

//! This module reexports all macros and types needed to define a quantity.

pub use crate::{Amnt, AmountT, Quantity, SIPrefix, Unit, Unitless};
#[cfg(feature = "fpdec")]
pub use crate::{Dec, Decimal};
pub use core::cmp::Ordering;
pub use core::fmt;
pub use core::ops::{Add, Div, Mul, Sub};
pub use qty_macros::quantity;
