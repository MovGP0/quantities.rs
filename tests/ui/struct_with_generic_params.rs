// ---------------------------------------------------------------------------
// Copyright:   (c) 2022 ff. Michael Amrhein (michael@adrhinum.de)
// License:     This program is part of a larger application. For license
//              details please read the file LICENSE.TXT provided together
//              with the application.
// ---------------------------------------------------------------------------
// $Source$
// $Revision$

use quantities::prelude::*;

/// Foo, a completely useless quantity
#[quantity]
#[ref_unit(A, "a", MEGA)]
#[unit(B, "b", 0.4)]
#[unit(C, "c", 0.01)]
struct Foo<T> {}

fn main() {}
