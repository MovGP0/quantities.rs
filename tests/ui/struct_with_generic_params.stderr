error: Given struct must not have generic parameters.
  --> tests/ui/struct_with_generic_params.rs:17:11
   |
17 | struct Foo<T> {}
   |           ^^^
   |
   = help: Use `struct Foo {};`
